assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -58,18 +58,18 @@ const moduleDefaults = {
 	dependants: [],
 	order: 10,
 	icon: null,
-	settingsComponent: DefaultModuleSettings,
+	settingsEditComponent: null,
+	settingsViewComponent: null,
 };
 
 const normalizeModules = memize(
 	( modules ) => Object.keys( modules )
 		.map( ( slug ) => {
-			return {
-				...moduleDefaults,
-				name: slug, // Ensure `name` is not empty.
-				...modules[ slug ],
-				slug,
-			};
+			const module = { ...modules[ slug ], slug };
+			// Fill any `undefined` values with defaults.
+			defaults( module, { name: slug }, moduleDefaults );
+
+			return module;
```

Comment:
Note this fixes a minor bug where defaults would be overwritten by an `undefined` value which will still take precedence when spreading into an object.